### PR TITLE
Assignment feature

### PR DESCRIPTION
**Pull request by BartDevTWG**
assignment feature 

Zrzut ekranu 2021-02-15 o 18 46 31
Zrzut ekranu 2021-02-15 o 18 47 21

## Checklist

- [ ] Lint and unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: src/auth/authReducer.ts [hunk 1]
```diff
@@ -10,8 +10,14 @@ export enum AuthStatus {
   ERROR = 'ERROR',
 }
 
+export interface MePayload {
```
Comment (pavelsof) [inline src/auth/authReducer.ts:13]: Nice.. I did not know this was not used until now.. :D

**Review by pavelsof — APPROVED** [review]
Nice!

By the way, I added the `all` inbox to the response of the [inbox list endpoint](https://github.com/cara-care/dashboard/wiki/Chat#rooms-and-inboxes), it does not have to be hardcoded any more (:

File: src/chat/__tests__/Inbox.test.tsx [hunk 2]
```diff
@@ -15,6 +18,14 @@ describe('<Inbox />', () => {
...
   beforeEach(() => {
     jest.clearAllMocks();
+    //@ts-ignore
+    mockGetInboxesList.mockResolvedValueOnce({
+      data: { results: chatConversationsMock },
+    });
+    //@ts-ignore
+    mockGetChatAuthorizationToken.mockResolvedValueOnce({
+      data: { token: 'test' },
+    });
```
Comment (MichaelAmoabengVaulitions) [inline src/chat/__tests__/Inbox.test.tsx:28]: Can you provide some helpful comment why TS is ignored here and how we can fix or prevent this in the future?

File: src/chat/__tests__/InboxSidebar.test.tsx [hunk 3]
```diff
@@ -1,13 +1,25 @@
...
+
+jest.mock('../../utils/api');
 
 describe('<InboxSidebar />', () => {
-  it('does render correctly', () => {
+  it('does render correctly', async () => {
+    // @ts-ignore
+    mockGetInboxesList.mockResolvedValueOnce({
+      data: { results: chatConversationsMock },
+    });
```
Comment (MichaelAmoabengVaulitions) [inline src/chat/__tests__/InboxSidebar.test.tsx:17]: Same here . reason for TS ignore

File: src/utils/api.ts [hunk 4]
```diff
@@ -157,18 +158,24 @@ export const changePassword = ({
...
+      : `/dashboard/chat/rooms/?inbox=${chatRoomsSlug}&limit=${limit}&offset=${offset}`;
+  return api.get(url);
 };
 
-export const getChatRoom = (_: string, userId: number) => {
-  return api.get(`/dashboard/chat/rooms/${userId}/`);
-};
+// export const getChatRoom = (_: string, userId: number) => {
+//   return api.get(`/dashboard/chat/rooms/${userId}/`);
+// };
```
Comment (MichaelAmoabengVaulitions) [inline src/utils/api.ts:179]: Why is this commented out? Maybe you can remove if its not needed or provide a reason why it might be useful

**Review by MichaelAmoabengVaulitions — CHANGES_REQUESTED** [review]
Nice work here @BartDevTWG 🤝. I left a few comments.

File: src/utils/api.ts (hunk 4 above)
Comment (BartDevTWG) [reply src/utils/api.ts:179]: It's no longer used, will remove it

File: src/chat/__tests__/Inbox.test.tsx (hunk 2 above)
Comment (BartDevTWG) [reply src/chat/__tests__/Inbox.test.tsx]: I removed ts-ignore by marking these functions as a jest.Mock :)

**Comment by BartDevTWG** [comment]
@MichaelAmoabengVaulitions Could you check again ?

**Review by MichaelAmoabengVaulitions — APPROVED** [review]
LGTM!

**Pull request merged**